src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/ConfiguredCancelableAsyncEnumerableTests.netcoreapp.cs

```diff
@@ -0,0 +1,216 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Sources;
+using Xunit;
+
+namespace System.Runtime.CompilerServices.Tests
+{
+    public class ConfiguredCancelableAsyncEnumerableTests
+    {
+        [Fact]
+        public void Default_GetAsyncEnumerator_Throws()
+        {
+            ConfiguredCancelableAsyncEnumerable<int> e = default;
+            Assert.Throws<NullReferenceException>(() => e.GetAsyncEnumerator());
+
+            e = ((IAsyncEnumerable<int>)null).ConfigureAwait(false);
+            Assert.Throws<NullReferenceException>(() => e.GetAsyncEnumerator());
+        }
+
+        [Fact]
+        public void Default_EnumeratorMembers_Throws()
+        {
+            ConfiguredCancelableAsyncEnumerable<int>.Enumerator e = default;
+            Assert.Throws<NullReferenceException>(() => e.MoveNextAsync());
+            Assert.Throws<NullReferenceException>(() => e.Current);
+            Assert.Throws<NullReferenceException>(() => e.DisposeAsync());
+        }
+
+        [Fact]
+        public void Default_WithCancellation_ConfigureAwait_NoThrow()
+        {
+            ConfiguredCancelableAsyncEnumerable<int> e = TaskExtensions.WithCancellation((IAsyncEnumerable<int>)null, default);
+            e = e.ConfigureAwait(false);
+            e = e.WithCancellation(default);
+            Assert.Throws<NullReferenceException>(() => e.GetAsyncEnumerator());
+        }
+
+        [Fact]
+        public void Default_ConfigureAwait_WithCancellation_NoThrow()
```

Comment:
>Default_ConfigureAwait_WithCancellation_NoThrow [](start = 20, length = 47)

Consider testing calling ConfigureAwait multiple times, and checking that the last value wins.
Same for cancellation token